src/main/java/org/elasticsearch/action/update/UpdateRequest.java

```diff
@@ -205,105 +201,16 @@ int shardId() {
-     */
-    public UpdateRequest script(String script, ScriptService.ScriptType scriptType) {
-        this.script = script;
-        this.scriptType = scriptType;
-        return this;
-    }
-
     /**
      * The script to execute. Note, make sure not to send different script each times and instead
      * use script params if possible with the same (automatically compiled) script.
      */
-    public UpdateRequest script(String script) {
+    public UpdateRequest script(Script script) {
         this.script = script;
-        this.scriptType = ScriptService.ScriptType.INLINE;
-        return this;
-    }
-
-
-    /**
-     * The language of the script to execute.
-     */
-    public UpdateRequest scriptLang(String scriptLang) {
-        this.scriptLang = scriptLang;
-        return this;
-    }
-
-    public String scriptLang() {
-        return scriptLang;
-    }
-
-    /**
-     * Add a script parameter.
-     */
-    public UpdateRequest addScriptParam(String name, Object value) {
-        if (scriptParams == null) {
-            scriptParams = Maps.newHashMap();
-        }
-        scriptParams.put(name, value);
-        return this;
-    }
-
-    /**
-     * Sets the script parameters to use with the script.
-     */
-    public UpdateRequest scriptParams(Map<String, Object> scriptParams) {
-        if (this.scriptParams == null) {
-            this.scriptParams = scriptParams;
-        } else {
-            this.scriptParams.putAll(scriptParams);
-        }
-        return this;
-    }
-
-    /**
-     * The script to execute. Note, make sure not to send different script each times and instead
-     * use script params if possible with the same (automatically compiled) script.
-     */
-    public UpdateRequest script(String script, ScriptService.ScriptType scriptType, @Nullable Map<String, Object> scriptParams) {
-        this.script = script;
-        this.scriptType = scriptType;
-        if (this.scriptParams != null) {
-            this.scriptParams.putAll(scriptParams);
-        } else {
-            this.scriptParams = scriptParams;
-        }
-        return this;
-    }
-
-    /**
-     * The script to execute. Note, make sure not to send different script each times and instead
-     * use script params if possible with the same (automatically compiled) script.
-     *
-     * @param script       The script to execute
-     * @param scriptLang   The script language
-     * @param scriptType   The script type
-     * @param scriptParams The script parameters
-     */
-    public UpdateRequest script(String script, @Nullable String scriptLang, ScriptService.ScriptType scriptType, @Nullable Map<String, Object> scriptParams) {
-        this.script = script;
-        this.scriptLang = scriptLang;
-        this.scriptType = scriptType;
-        if (this.scriptParams != null) {
-            this.scriptParams.putAll(scriptParams);
-        } else {
-            this.scriptParams = scriptParams;
-        }
```

Comment:
As above I kept UpdateRequestBuilder bwc as I thought that was the bit exposed for the Java API